Library/Homebrew/formula_installer.rb

```diff
@@ -942,13 +943,30 @@ def post_install
     @show_summary_heading = true
   end
 
-  def pour
+  def fetch
+    return if @fetched
```

Comment:
You might want to move this duplicate fetch check into https://github.com/Homebrew/brew/blob/cc05c3df5cf94b7607e56d3cf76d404244363e4e/Library/Homebrew/download_strategy.rb#L284-L329

Currently because `fetch_bottle` still can call `downloader.fetch` you'll get the duplicated output. You could have a `@fetch_message_printed` or similar and use that to avoid printing the message but otherwise have the behaviour be identical.

Reply:
I gave this a whirl, adding only `@fetch_message_printed` while leaving the behavior unchanged. 

```
==> Downloading https://curl.haxx.se/download/curl-7.69.1.tar.bz2
Already downloaded: /Users/avocado/Library/Caches/Homebrew/downloads/be9bfe94edb7fc0d0c87d756311054a8c8dabc6d97cdb9a7c12c078a569d9b72--curl-7.69.1.tar.bz2
#<CurlDownloadStrategy:0x00007fcdbe2e67e0>
==> Downloading https://curl.haxx.se/download/curl-7.69.1.tar.bz2
Already downloaded: /Users/avocado/Library/Caches/Homebrew/downloads/be9bfe94edb7fc0d0c87d756311054a8c8dabc6d97cdb9a7c12c078a569d9b72--curl-7.69.1.tar.bz2
#<CurlDownloadStrategy:0x00007f8d23abd008>
```

I printed out the `download_strategy` each time and it appears the downloaders are two different instances as `build` happens in a different process. 

It is possible to pass in an argument to `download_strategy`, though I am not sure if that is the most eloquent approach. 

Any opinions or alternatives?

Reply:
> I printed out the `download_strategy` each time and it appears the downloaders are two different instances as `build` happens in a different process.

Doh! 🤦‍♂ 

> It is possible to pass in an argument to `download_strategy`, though I am not sure if that is the most eloquent approach.

That seems good to me! 👍